### ISSUE TITLE
request: torch.compile support for tensor subclass subgraph boundaries

### ISSUE THREAD

**Issue by vkuzo** [pytorch/pytorch#117115]
### 🐛 Describe the bug

Is it possible to enable tensor subclasses in subgraph boundaries?  It doesn't seem to currently work for `Float8Tensor`.  Logs: https://gist.github.com/vkuzo/53e92dfd91879c0710975b8a657fd91d

This is needed for easier development of float8 training (currently we are forced to keep everything full graph compatible, this is really restrictive for experimental projects).

Full reproduction instructions: https://github.com/pytorch-labs/float8_experimental/pull/166

### Versions

```
(pytorch) [vasiliy@devgpu009.ftw2 ~/local/pytorch (main)]$ python
Python 3.11.4 (main, Jul  5 2023, 13:45:01) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch; torch.__version__
'2.2.0a0+git967863d'
```

cc @Chillee @ezyang @albanD @samdow @chauhang @penguinwu @bdhirsh @bobrenjc93 @aorenste @zou3519 @msaroufim @wconstab @anijain2305

**Comment by vkuzo** [comment]
cc @bdhirsh , @voznesenskym , thoughts on who could help figure out how to make this work?

**Comment by bdhirsh** [comment]
FWIW: tensor subclasses at graph boundaries works today, see https://github.com/pytorch/pytorch/blob/main/test/functorch/test_aotdispatch.py#L3673

There is probably a specific bug at the intersection of graph breaks / the particular autograd.Function-y code in the float8 codebase / the float8 subclass itself. I can try to take a look in the next few days

**Comment by zou3519** [comment]
We're scrubbing old issues. Does this issue still apply? @bdhirsh

**Comment by vkuzo** [comment]
yes, this is important for low precision training and inference, which depends on torch.compile for performance.  Without this, users are forced to write their code without graph breaks which is often challenging and leads to a painful devex.
